ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureClient.java

```diff
@@ -32,18 +34,77 @@
     private final CardinalClient cardinalClient;
     private final BraintreeClient braintreeClient;
     private final ThreeDSecureV1BrowserSwitchHelper browserSwitchHelper;
+    private final ThreeDSecureAPI api;
+    private ThreeDSecureListener listener;
 
+    @VisibleForTesting
+    ThreeDSecureLifecycleObserver observer;
+
+    /**
+     * Create a new instance of {@link ThreeDSecureClient} from within an Activity using a {@link BraintreeClient}.
+     *
+     * @param activity a {@link FragmentActivity}
+     * @param braintreeClient a {@link BraintreeClient}
+     */
+    public ThreeDSecureClient(@NonNull FragmentActivity activity, @NonNull BraintreeClient braintreeClient) {
+        this(activity, activity.getLifecycle(), braintreeClient, new CardinalClient(), new ThreeDSecureV1BrowserSwitchHelper(), new ThreeDSecureAPI(braintreeClient));
+    }
+
+    /**
+     * Create a new instance of {@link ThreeDSecureClient} from within a Fragment using a {@link BraintreeClient}.
+     *
+     * @param fragment a {@link Fragment}
+     * @param braintreeClient a {@link BraintreeClient}
+     */
+    public ThreeDSecureClient(@NonNull Fragment fragment, @NonNull BraintreeClient braintreeClient) {
+        this(fragment.requireActivity(), fragment.getLifecycle(), braintreeClient, new CardinalClient(), new ThreeDSecureV1BrowserSwitchHelper(), new ThreeDSecureAPI(braintreeClient));
+    }
+
+    /**
+     * Create a new instance of {@link ThreeDSecureClient} using a {@link BraintreeClient}.
+     *
+     * Deprecated. Use {@link ThreeDSecureClient(Fragment, BraintreeClient)} or
+     * {@link ThreeDSecureClient(FragmentActivity, BraintreeClient)} instead.
+     *
+     * @param braintreeClient a {@link BraintreeClient}
+     */
+    @Deprecated
```

Comment:
I ended up deprecating the old integration pattern to make it more clear which methods support the new pattern. We can revisit whether or not we want to deprecate these methods yet, though. I just wasn't sure how to make it clear which methods go together without deprecating.